script/hardhat/fork/vaultUpgradesVip.ts

```diff
@@ -0,0 +1,333 @@
+        params: [VRT_VAULT_PROXY, "pause()", FAST_TRACK_TIMELOCK],
+      },
+
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [VRT_VAULT_PROXY, "pause()", CRITICAL_TIMELOCK],
+      },
+
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [VRT_VAULT_PROXY, "resume()", FAST_TRACK_TIMELOCK],
+      },
+
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [VRT_VAULT_PROXY, "resume()", CRITICAL_TIMELOCK],
+      },
+    ],
+    meta,
+    ProposalType.REGULAR,
+  );
+};
+
+forking(27077371, async () => {
+  let xvsVaultProxy: XVSVaultProxy;
+  let vrtVaultProxy: VRTVaultProxy;
+  let xvsVault: XVSVault;
+  let vaiVault: VAIVault;
+  let vrtVault: VRTVault;
+  let accessControlManager: IAccessControlManager;
+  let xvsVaultSigner;
+  let vaiVaultSigner;
+  let vrtVaultSigner;
+
+  before(async () => {
+    xvsVault = await ethers.getContractAt("XVSVault", XVS_VAULT_PROXY);
+    vaiVault = await ethers.getContractAt("VAIVault", VAI_VAULT_PROXY);
+    vrtVault = await ethers.getContractAt("VRTVault", VRT_VAULT_PROXY);
+
+    xvsVaultProxy = await ethers.getContractAt("XVSVaultProxy", XVS_VAULT_PROXY);
+    vrtVaultProxy = await ethers.getContractAt("VRTVaultProxy", VRT_VAULT_PROXY);
+    const multiSigSigner = await initMainnetUser(MULTISIG, ethers.utils.parseEther("1.0"));
+
+    await xvsVaultProxy.connect(multiSigSigner)._setPendingAdmin(NORMAL_TIMELOCK);
+    await vrtVaultProxy.connect(multiSigSigner)._setPendingAdmin(NORMAL_TIMELOCK);
+
+    xvsVaultSigner = await initMainnetUser(XVS_VAULT_PROXY, ethers.utils.parseEther("1"));
+    vaiVaultSigner = await initMainnetUser(VAI_VAULT_PROXY, ethers.utils.parseEther("1"));
+    vrtVaultSigner = await initMainnetUser(VRT_VAULT_PROXY, ethers.utils.parseEther("1"));
+    accessControlManager = await ethers.getContractAt("IAccessControlManager", ACM);
+  });
+
+  describe("Pre-VIP behavior", async () => {
+    it("Owner of XVSVault is Multisig", async () => {
+      const owner = await xvsVault.admin();
+      expect(owner).to.equal(MULTISIG);
+    });
+
+    it("Owner of VRTVault is Multisig", async () => {
+      const owner = await vrtVault.admin();
+      expect(owner).to.equal(MULTISIG);
+    });
+
+    it("Implementation of XVSVault", async () => {
+      const impl = await xvsVault.implementation();
+      expect(impl).to.equal(XVS_OLD);
+    });
+
+    it("Implementation of VAIVault", async () => {
+      const impl = await vaiVault.vaiVaultImplementation();
+      expect(impl).to.equal(VAI_OLD);
+    });
+
+    it("Implementation of VRTVault", async () => {
+      const impl = await vrtVault.implementation();
+      expect(impl).to.equal(VRT_OLD);
+    });
+  });
+
+  testVip("VIP-105 Change Vault Implementation", vip105());
+
+  describe("Post-VIP behavior", async () => {
```

Comment:
Don't we have to pretend the execution of the VIP before this batch of checks? @aditya-routv

Reply:
I see, thanks!